projects/core/src/routing/event/routing.events.ts

```diff
@@ -0,0 +1,29 @@
+import { Category, Price } from '../../model/product.model';
+
+export class HomePageVisited {}
+export class CartPageVisited {}
+export class OrderConfirmationPageVisited {}
+export class PageVisited {}
+
+/**
+ * Indicates that a user visited a PDP page.
+ * A visited product code value is emitted whenever the PDP page is visited.
+ */
+export class ProductDetailsPageVisited {
+  categories?: Category[];
+  code?: string;
+  name?: string;
+  price?: Price;
+}
+/**
+ * Indicates that a user visited a category or brand page.
+ */
+export class CategoryPageVisited {
+  categoryCode: string;
+  categoryName: string;
+}
+
+export class KeywordSearchPageVisited {
```

Comment:
Can we rename this event to just `SearchPageVisited`?

Reply:
In our world we always use the term KeywordSearch. Is it a problem or can we leave it like this?

Reply:
renamed